File: benchmarks/src/main/resources/js/primenumbers.array.js
function getNthPrime(n) {
  if (n === 1) {
    return 2
  }
  if (n === 2) {
    return 3
  }

  var primes = [];
  primes.push(2);
  primes.push(3);

  var primeCounter = 2;
  var candidate = 5;
  while (primeCounter < n) {
    var isPrime = true;
    var dividerIndex = 0;
    while (isPrime) {
      var divider = primes[dividerIndex];
      if (divider * divider > candidate) {
        break;
      }

      if (candidate % divider === 0) {
        isPrime = false;
      }
      dividerIndex++;
    }
    if (isPrime) {
      primes.push(candidate);
      ++primeCounter;
    }
    candidate += 2;
  }
  return primes[primes.length - 1];
}

File: benchmarks/src/main/resources/python/primenumbers_array.py
def get_nth_prime(n):
  if n == 1:
    return 2
  if n == 2:
    return 3
  primes = [2.0, 3.0]
  found = 2
  candidate = 5.0
  while found < n:
    is_prime = True
    for prime in primes:
      if prime * prime > candidate:
        break

      if candidate % prime == 0:
        is_prime = False
        break

    if is_prime:
      primes.append(candidate)
      found += 1

    candidate += 2

  return primes[n - 1]